tests/utils/test_compression.py

```diff
@@ -54,8 +54,8 @@ def setUp(self):
                 f_bz2.writelines([header, line1, line2])
 
         # Base Exception so it catches Keyboard Interrupt
-        except BaseException as e:
-            logging.error(e)
+        except BaseException:
+            logging.exception("An exception has occurred.")
```

Comment:
The flake8-logging-format extension isn't fond of using the exception itself in the logging message.

`G200 Logging statements should not include the exception in logged string (use exception or exc_info=True)`

Reply:
But perhaps going this route for core and provider modules, we get into a situation where log handlers don't output the same (possibly useful?) information. I'd love feedback here.

Reply:
And here too :)

Reply:
Shuffled some pieces around and now this test is using a fixture for teardown activity + converted to `pytest` since I was messing around anyway.